src/Api/ProductApi.php

```diff
@@ -108,8 +108,8 @@ public function delete(string $code): int
     /**
      * {@inheritdoc}
      */
-    public function upsertList($products): \Traversable
+    public function upsertList($resources): \Traversable
     {
-        return $this->resourceClient->upsertStreamResourceList(static::PRODUCTS_URI, [], $products);
+        return $this->resourceClient->upsertStreamResourceList(static::PRODUCTS_URI, [], $resources);
```

Comment:
Why switch to the generic naming? I think keeping the context is fine.
Also if we change here I think we should change everywhere (see AttributeApi, CategoryApi etc)

Reply:
A hint of my IDE: `parameter's name changed from 'resources' to 'attributes' ` because, in the `UpsertableResourceListInterface`, we have  `public function upsertList($resources): \Traversable;`. You're right, the mistake is also in other files, so I have to change everywhere or nowhere ^^